src/components/app/ProfileDeleteButton.js

```diff
@@ -25,14 +25,83 @@ type Props = {|
   +onCloseSuccessMessage?: () => mixed,
 |};
 
-type State = {|
-  +status: 'idle' | 'working' | 'just-deleted' | 'deleted',
+export class ProfileDeleteButton extends PureComponent<ButtonProps> {
+  _hasBeenDeleted = false;
+  _componentDeleteButtonRef = React.createRef<ButtonWithPanel>();
+
+  onCloseConfirmDialog = () => {
+    // In case we deleted the profile, and the user dismisses the success panel,
+    // let's move directly to the deleted state:
+    if (this._hasBeenDeleted && this.props.onCloseSuccessMessage) {
+      this.props.onCloseSuccessMessage();
+    }
+
+    if (this.props.onCloseConfirmDialog) {
```

Comment:
Probably not a conscious choice. It's common that I make callbacks optional by default, because I consider that they're not necessary for the component to work. But I see I made the callbacks for the Panel non-optional so it's not even consistent on my side 😎.

Also initially I thought I could reuse this button directly and so I believe I wanted to make it more generic. Now it's used only in one location and won't probably be reused elsewhere, so I could definitely make them non-optional as part of this refactoring. Later we could maybe move the "button" part to `ListOfPublishedProfiles`.